alpha/declcfg/write.go

```diff
@@ -109,31 +161,50 @@ func WriteMermaidChannels(cfg DeclarativeConfig, out io.Writer, minEdgeName stri
 		out.Write([]byte(pkgs[pkgName].String()))
 		out.Write([]byte("  end\n"))
 	}
-	out.Write([]byte("<!-- PLEASE NOTE:  skipRange edges are not currently displayed -->\n"))
 
 	return nil
 }
 
 // filters the channel edges to include only those which are greater-than-or-equal to the edge named by startVersion
 // returns a nil channel if all edges are filtered out
-func filterChannel(c *Channel, versionMap map[string]semver.Version, minEdgeName string) *Channel {
-	// short-circuit if no specified startVersion
-	if minEdgeName == "" {
+func (writer *MermaidWriter) filterChannel(c *Channel, versionMap map[string]semver.Version, minVersion semver.Version) *Channel {
+	// short-circuit if no active filters
+	if writer.MinEdgeName == "" && writer.SpecifiedPackageName == "" {
 		return c
 	}
-	// convert the edge name to the version so we don't have to duplicate the lookup
-	minVersion := versionMap[minEdgeName]
+
+	// short-circuit if channel's package doesn't match filter
+	if writer.SpecifiedPackageName != "" && c.Package != writer.SpecifiedPackageName {
+		return nil
+	}
+
+	// short-circuit where the minEdgeName is not in this channel
```

Comment:
Not sure if this is really a "short" short-circuit. It's a lot of work for a check isn't it? We're looping on `c.Entries` right below as well, can't any checking for the MinEdgeName just be combined into that loop?

Reply:
It evaluates a single criterion across the range of entries in the channel in order to recognize a case where there is nothing pertinent in the channel (when the minimumEdge is specified).  

I had attempted to include this in the logic of the larger loop, but it complicated the logic something fierce, as well as introducing inefficiencies like having to parse the skipRange twice for validation.  

I'll take a look at it though to see if I can smooth it out.  If you have any ideas, I'd love to hear them as well.